### PR TITLE
Newsletters: add 273 (2023-10-18)

### PR DESCRIPTION
**Pull request by harding**
- [x] PSBT BIP news @harding 
- [x] Other merged summaries @harding 
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Client/Services @bitschmidty 
- [x] Bitcoin Core 27255 @murchandamus

### CODE REVIEW

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    channel between Alice and Bob, multiple programs could be run in
+    both parallel and in sequence with no onchain footprint except
+    channel setup and either a mutual close or a force close where Bob
+    attempts to demonstrate that Alice failed to follow the arbitrary
+    program logic correctly.
+
+    BitVM can be applied trustlessly in cases where Alice and Bob are
+    natural adversaries, such as where they pay funds to an output that
+    will be paid to whichever one of them wins in a game of chess.  They
+    can then use an two (almost identical) arbitrary programs, each of
```
Comment (vostrnad) [inline _posts/en/newsletters/2023-10-18-newsletter.md:56]:
```suggestion
    can then use two (almost identical) arbitrary programs, each of
```

**Review by murchandamus — COMMENTED** [review]
LGTM so far

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME
+
+## News
+
+- **Payments contingent on arbitrary computation:** Robin Linus
+  [posted][linus post] to the Bitcoin-Dev a [paper][linus paper] he's
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:15]:
```suggestion
  [posted][linus post] to the Bitcoin-Dev  mailing list a [paper][linus paper] he's
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    Linus's paper gets around some of those limits.  If Alice trusts Bob
+    to take action if a program was run incorrectly, but does not want
+    to trust him with anything else, she can pay funds to a taproot tree
+    that will allow Bob to claim the funds if he demonstrates that Alice
+    failed to run an arbitrary program correctly.  If Alice does run the
+    program correctly, then she can spend the funds even if Bob attempts
+    to stop her.
+
+    To use an arbitrary program, it must be broken down into a very
+    basic primitive (a NAND gate) and a commitment must be made for each
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:39]: This link (or something better) could be useful. Not strong on it.:
https://en.wikipedia.org/wiki/NAND_gate

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    gate.  This requires the offchain exchange of a very large amount of
+    data, potentially multiple gigabytes for even a fairly basic
+    arbitrary program---but Alice and Bob only need a single onchain
+    transaction in the case that Bob agrees that Alice ran the program
+    correctly.  In the case that Bob disagrees, he should be able to
+    demonstrate Alice's failure within a relatively small number of
+    onchain transactions.  If the setup was performed in a payment
+    channel between Alice and Bob, multiple programs could be run in
+    both parallel and in sequence with no onchain footprint except
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:48]: Feels like to me this wants to read 
"could be run both in parallel and in sequence with no onchain"

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo], and
+[Bitcoin Inquisition][bitcoin inquisition repo].*
+
+- [Bitcoin Core #27255][] ports Miniscript to Tapscript. This code change makes
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:129]:
```suggestion
- [Bitcoin Core #27255][] ports miniscript to [tapscript][topic tapscript]. This code change makes
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  Miniscript an option for P2TR output descriptors, adding support for both
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:130]:
```suggestion
  miniscript an option for P2TR [output descriptors][topic descriptors], adding support for both
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    order to spend bitcoins associated with that script.  For example, a
+    script containing a public key that can only be satisfied if the
+    corresponding private key creates a signature committing to a
+    spending transaction.  Scripts must be written in Bitcoin's language
+    (called _Script_) in order to be enforced by consensus, but Script
+    is deliberately limited in its flexibility.
+
+    Linus's paper gets around some of those limits.  If Alice trusts Bob
+    to take action if a program was run incorrectly, but does not want
+    to trust him with anything else, she can pay funds to a taproot tree
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:32]:
```suggestion
    to trust him with anything else, she can pay funds to a [taproot][topic taproot] tree
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo], and
+[Bitcoin Inquisition][bitcoin inquisition repo].*
+
+- [Bitcoin Core #27255][] ports Miniscript to Tapscript. This code change makes
+  Miniscript an option for P2TR output descriptors, adding support for both
+  watching and signing "TapMiniscript descriptors". Previously, Miniscript was
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:131]:
```suggestion
  watching and signing "TapMiniscript descriptors". Previously, miniscript was
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+  available only for P2WSH output descriptors. The author notes that a new
+  `multi_a` fragment is introduced exclusively for P2TR descriptors that
+  matches the semantics of `multi` in P2WSH descriptors. The discussion on the
+  PR notes that a majority of the work went towards proper tracking of the
+  changed resource limits for Tapscript.
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-10-18-newsletter.md:136]:
```suggestion
  changed resource limits for tapscript.
```

**Comment by murchandamus** [comment]
Force pushed update to fix style and add topic links suggested by @bitschmidty

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+    either accept that claim (conceding the loss of funds) or will
+    demonstrate the fraud (receiving the funds if successful).  In cases
+    where Alice and Bob would not naturally be adversaries, Alice may be
+    able to incentivize him to verify correct computation, such by
+    offering her funds to Bob if he can prove she failed to compute
+    correctly.
+
+    The idea received a significant amount of discussion on the mailing
+    list as well as on Twitter and various Bitcoin-focused podcasts.  We
+    expect significant continued discussion in the coming weeks and
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-10-18-newsletter.md:70]:
```suggestion
    expect substantial continued discussion in the coming weeks and
```
(Nit, just so we're not using "significant" twice close together. Or drop this word.)

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+  The firmware [updates][coldcard blog] include [BIP370][] support for
+  version 2 [PSBTs][topic psbt], additional [BIP39][] support, and multiple seed capabilities.
+
+- **Tapleaf circuits: a BitVM demo:**
+  [Tapleaf circuits][] is a proof-of-concept implementation of Bristol circuits
+  using the BitVM approach outlined earlier in the newsletter.
+
+- **Samourai Wallet 0.99.98i released:**
+  The [0.99.98i][samourai blog] release includes additional PSBT, UTXO labeling,
+  and batch sending features.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-10-18-newsletter.md:99]:
```suggestion
  and batch-sending features.
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+- **Krux: signing device firmware:**
+  [Krux][krux github] is an open source firmware project for building hardware
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-10-18-newsletter.md:102]:
```suggestion
  [Krux][krux github] is an open-source firmware project for building hardware
```

**Comment by harding** [comment]
Added missing content and reviewed @murchandamus and @bitschmidty contributions (thanks!).

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+slug: 2023-10-18-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter briefly mentions a recent security disclosure
+affecting LN users and describes a paper about making payments
+contingent on the result of running arbitrary programs.  Also included
+are our regular sections that summarize improvements to clients and
+services, announce new release and release candidates, and describe
```
Comment (vostrnad) [inline _posts/en/newsletters/2023-10-18-newsletter.md:14]:
```suggestion
services, announce new releases and release candidates, and describe
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+  signing devices using commodity hardware.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Bitcoin Core 24.2rc2][] and [Bitcoin Core 25.1rc1][] are release
+  candidates for a maintenance versions of Bitcoin Core.
```
Comment (vostrnad) [inline _posts/en/newsletters/2023-10-18-newsletter.md:128]:
```suggestion
  candidates for maintenance versions of Bitcoin Core.
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+  PR notes that a majority of the work went towards proper tracking of the
+  changed resource limits for tapscript.
+
+- [Eclair #2703][] discourages spenders from forwarding payments through
+  the local node when the node's balance is low and it would probably
+  need to reject those payments.  This is accomplished by the node
+  advertising that its maximum HTLC amount has been lowered.  Preventing
+  rejected payments improves the experience for spenders and helps avoid
+  the local node being penalized by pathfinding systems that discount
+  nodes the have failed to forward a payment in the recent past.
```
Comment (jirijakes) [inline _posts/en/newsletters/2023-10-18-newsletter.md:155]:
```suggestion
  nodes that have failed to forward a payment in the recent past.
```

File: _posts/en/newsletters/2023-10-18-newsletter.md [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+name: 2023-10-18-newsletter
+slug: 2023-10-18-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter briefly mentions a recent security disclosure
+affecting LN users, describes a paper about making payments
+contingent on the result of running arbitrary programs, and announces a
+proposed BIP to fields to PSBTs for MuSig2.  Also included
```
Comment (vostrnad) [inline _posts/en/newsletters/2023-10-18-newsletter.md:13]:
```suggestion
proposed BIP to add fields to PSBTs for MuSig2.  Also included
```

**Comment by bitschmidty** [comment]
Accepted a few last minute suggestions from review. Thank you @jirijakes and @vostrnad ! And took @LarryRuane 's suggestions for the client/services section.

Going to publish as is, but considerations for follow up topic entries:
- Responsible disclosures for the LN vulnerability (or when we publish more details next week)
- MuSig2 / PSBTs for the proposed bip
- miniscript/tapscript for Bitcoin Core #27255

Squashed and merged, thanks all! 🚀

**Pull request merged**